modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -12106,4 +12106,33 @@ public static long uncompressedSize(File zip) throws IOException {
             return size;
         }
     }
+
+    /**
+     * Map object hash to some index between 0 and specified size via modulo operation.
+     *
+     * @param hash Object hash.
+     * @param size Size greater than 0.
+     * @return Calculated index in range [0..size).
+     */
+    public static int hashToIndex(int hash, int size) {
+        return safeAbs(hash % size);
+    }
+
+    /**
+     * Map object hash to some index between 0 and specified size using
+     * approach similar to {@link HashMap#hash(Object)}. In case
+     * <code>size</code> is not a power of 2 we fallback to modulo operation.
+     *
+     * @param hash Object hash.
+     * @param size Size greater than 0.
+     * @return Calculated index in range [0..size).
+     */
+    public static int hashToIndexOpt(int hash, int size, boolean isSizePow2) {
```

Comment:
Well, I don't think that you need to provide a unique name here, overloading is enough.